### ISSUE TITLE
DWIN_CREALITY_LCD

### ISSUE THREAD

**Issue by ffnull** [CrealityOfficial/Sermoon-V1#1]
Hello
First of all. Thnx for source codes.
But sources for LCD is missing. Please update sources to include DWIN display sources of our printer. Thnx.

**Comment by adamalfath** [comment]
+1
Would really appreciate if Creality could share the DWIN project file.

**Comment by danilodorgam** [comment]
+1
Would really appreciate if Creality could share the DWIN project file.

**Comment by Stevetm2** [comment]
+1
Would also really appreciate the DWIN source code.  Unless you plan to add z-offset control during print and other UI features which the community are requesting?  Thanks!

**Comment by adamalfath** [comment]
Hi @ffnull, @danilodorgam & @Stevetm2, I've uploaded Sermoon V1 HMI project file in my repo [here](https://github.com/adamalfath/Sermoon-V1-LCD). Turns out it's quite easy to get the original project files in DGUS, just import the official DWIN_SET folder to a newly created project in DGUS, thanks to this [guide](https://github.com/juliandroid/DWIN_CR_10s_Pro). Shame that Creality didn't reply to the user query, even to just confirm if they're able to give it or not.

I've also made some minor QOL update both to [firmware](https://github.com/adamalfath/Sermoon-V1) & [display ](https://github.com/adamalfath/Sermoon-V1-LCD/tree/modded)if you want to take a look.

Cheers,

**Comment by Stevetm2** [comment]
This is great @adamalfath I'll be taking a look at this!  You may see that I falked from your code a few weeks ago.  I just submitted a back port of the linear advance fixes from Marlin v2.1.2.  The fixes allow the steppers to work correctly on the V1, without failing to extrude on some models.  Give it a go, the new command is M900 and you set the K value based on the Marlin L.A. tutorial.  I was very happy when it worked out, it improves corner acuracy quite a bit.  I set K=0.03 as a starting point, via a custom USB cable connected to an old phone running octo4a.

**Comment by adamalfath** [comment]
@Stevetm2 Awesome update!
Though unfortunately I won't be able to use that. I've converted my Sermoon V1 to run on Klipper. It exposes me to more tuning parameters that can be easily accessed and re-tuned, vs re-compile method with Marlin. I've implemented ABL, pressure advance & input shaping.

Here's a [video](https://youtu.be/qI7M8NLOahg) I took at the very first print test. I'm very happy with the result.
"Draft" print mode is happily sits at 200mm/s & 2000mm/s^2 with a decent acceptable result. Although I use it conservatively at 75-100mm/s for casual printing with great results.

**Comment by Stevetm2** [comment]
Thats cool that you have Klipper running, I'm tempted to try that but I'm enjoying using the Creality Cloud app on my phone to slice and wifi print.  Would you mind if I shared your video to the Sermoon V1 Facebook group as they might have some interest in it?
Thats cool that you're printing nice and fast with success.  I've been printing at 120mm/s at 2200m/s2 for outer walls and 2500m/s2 for everything else with success for most print.  I increased the mm/s2 to 6000 for a test but I thought the printer was going to explode so I didn't do that again, lol.
I'm curious, how is the ABL working out, are you getting better leveling?  Also I noticed that you must also be changing the leveling screws, did that also help a lot?  How many ABL locations do you use btw?
I noticed you're using the stock fan and duct.  I designed an new duct to accommodate a 5015 fan here btw : https://www.thingiverse.com/thing:5868023

**Comment by adamalfath** [comment]
@Stevetm2 Yeah sure go ahead, and maybe I already join there since there's the only Sermoon group available there, I think.
Also we can continue discussing further there, in order not to hijack this original issue ticket topic.

**Comment by Stevetm2** [comment]
Yeah you're right, totally hijacking this thread, haha...  thanks again for the great info though!!

**Comment by ffnull** [comment]
> @Stevetm2 Awesome update! Though unfortunately I won't be able to use that. I've converted my Sermoon V1 to run on Klipper. It exposes me to more tuning parameters that can be easily accessed and re-tuned, vs re-compile method with Marlin. I've implemented ABL, pressure advance & input shaping.
> 
> Here's a [video](https://youtu.be/qI7M8NLOahg) I took at the very first print test. I'm very happy with the result. "Draft" print mode is happily sits at 200mm/s & 2000mm/s^2 with a decent acceptable result. Although I use it conservatively at 75-100mm/s for casual printing with great results.

You can make few jumpers on board and all fans on printer will be controllable and not so noizy

**Comment by adamalfath** [comment]
@ffnull Yep I already try that, except for the hotend fan and that's is the noisiest one. I could live with that since the printer is far from my work area, and let the fan on the PSU and electronic area at full blast. The noise level is around 45-50dB and 65dB at fast movement.
I utilize the unused fan port on the motherboard as a buzzer pin instead. Buzzer at 24V is very adequate to send auditory notifications, which I more prefer.

**Comment by ffnull** [comment]
> @Stevetm2 Awesome update! Though unfortunately I won't be able to use that. I've converted my Sermoon V1 to run on Klipper. It exposes me to more tuning parameters that can be easily accessed and re-tuned, vs re-compile method with Marlin. I've implemented ABL, pressure advance & input shaping.
> 
> Here's a [video](https://youtu.be/qI7M8NLOahg) I took at the very first print test. I'm very happy with the result. "Draft" print mode is happily sits at 200mm/s & 2000mm/s^2 with a decent acceptable result. Although I use it conservatively at 75-100mm/s for casual printing with great results.

@adamalfath I see that you use KlipperScreen with hyperpixel. did you rotate image?

**Comment by adamalfath** [comment]
> @adamalfath I see that you use KlipperScreen with hyperpixel. did you rotate image?

That's one is from Waveshare. I haven't found the right method to rotate the screen yet, each tutorial give a different result to my screen. Though it is annoying, I'll leave the screen in landscape orientation for now until I dig in again someday.

**Comment by ffnull** [comment]
> > @adamalfath I see that you use KlipperScreen with hyperpixel. did you rotate image?
> 
> That's one is from Waveshare. I haven't found the right method to rotate the screen yet, each tutorial give a different result to my screen. Though it is annoying, I'll leave the screen in landscape orientation for now until I dig in again someday.

@adamalfath Can you share adapter for raspberry and waveshare lcd? I had same kit

**Comment by ffnull** [comment]
@adamalfath im mean stl for mounting waveshare in place

**Comment by ffnull** [comment]
@adamalfath nevermind. already installed both. single problam - rotate klipperscreen

**Comment by adamalfath** [comment]
@ffnull I've uploaded my model for the [CR-Touch](https://grabcad.com/library/creality-sermoon-v1-cr-touch-mount-1) mount and [new LCD](https://grabcad.com/library/creality-sermoon-v1-4-3-waveshare-lcd-mount-1). I don't save the file for mounting for raspi, it's just a simple riser that match the PCB mounting hole and added a double tape to it (instead of directly to the board PCB).

Please let me know if you figure out how to rotate the screen :grinning:

**Comment by fierce1509** [comment]
Hi guys do any of you have the firmware for the V1 Pro which  will allow me to go up to 300 degrees? i have an all metal `hotend` on my sermoon v1 pro now and when i try to compile the firmware just get errors . Could I get  some help please? Thanks.

**Comment by Jahdeg** [comment]
> This is great [@adamalfath](https://github.com/adamalfath) I'll be taking a look at this! You may see that I falked from your code a few weeks ago. I just submitted a back port of the linear advance fixes from Marlin v2.1.2. The fixes allow the steppers to work correctly on the V1, without failing to extrude on some models. Give it a go, the new command is M900 and you set the K value based on the Marlin L.A. tutorial. I was very happy when it worked out, it improves corner acuracy quite a bit. I set K=0.03 as a starting point, via a custom USB cable connected to an old phone running octo4a.

@Stevetm2 How did you enable Linear Advance? I downloaded your code and uploaded it to my Sermoon V1 but it looks like its not working on OrcaSlicer PA test

**Comment by Stevetm2** [comment]
Sorry I've not looked at that code for a long time.  It should just work in orca if you have it set to Marlin.  However, I moved over to Klipper using the Sonic Pad and was able to get the Creality Cloud app working on it too.  I would totally recommend going the Klipper route by following the tutorials, it's worth it.

**Comment by Jahdeg** [comment]
> Sorry I've not looked at that code for a long time. It should just work in orca if you have it set to Marlin. However, I moved over to Klipper using the Sonic Pad and was able to get the Creality Cloud app working on it too. I would totally recommend going the Klipper route by following the tutorials, it's worth it.

Thanks for the response back! Yeah, that's what I was thinking but it's still not working as intended. I will look to Klipper in the future, right now ill stick with Marlin.
